Scripts/Source/minai_Sex.psc

```diff
@@ -172,273 +166,46 @@ Function StartSexScene(actor[] actors, bool bPlayerInScene, string tags="")
-          ;if OActor.VerifyActors(ostimActors)
-          StartSexScene(ostimActors, bPlayerInScene, tags)
-          ;EndIf
-        else
-          ; Target 1 is already in an OStim thread
-          Main.Debug("OStim detects target 1 is already in thread: " + ActiveOstimThreadID)
-          ostimActors = OThread.GetActors(ActiveOstimThreadID)
-          ; add akSpeaker to OStim actor array
-          ostimActors = PapyrusUtil.PushActor(ostimActors,akSpeaker)
-          ostimActors = OActorUtil.Sort(ostimActors, OActorUtil.EmptyArray())
-          Main.Debug("OStim added akSpeaker to array and sorted: " + ostimActors)
-          OThread.Stop(ActiveOstimThreadID)
-          Utility.Wait(2)
-          StartSexScene(ostimActors, bPlayerInScene, tags)
-        EndIf
-      EndIf
-    else
-      ; akSpeaker is already in an OStim thread
-      ostimActors = OThread.GetActors(ActiveOstimThreadID)
-      Main.Debug("Searching for random " + tags + " scene.")
-      string newScene = OLibrary.GetRandomSceneWithAnyActionCSV(ostimActors, tags)
-      Utility.Wait(0.5)
-      if newScene == ""
-        newScene = OLibrary.GetRandomSceneWithAnySceneTagCSV(ostimActors, tags)
-        Utility.Wait(1)
-      EndIf
-      Main.Debug("Ostim scene transition to: " + newScene + " for OStim Thread [" + ActiveOstimThreadID + "].")
-      if OThread.IsRunning(ActiveOstimThreadID)
-        OThread.NavigateTo(ActiveOstimThreadID, newScene)
-        if OThread.IsInAutoMode(ActiveOstimThreadID)
-          OThread.StopAutoMode(ActiveOstimThreadID)
-          Utility.Wait(5)
-          OThread.StartAutoMode(ActiveOstimThreadID)
-        EndIf
-      EndIf
-      ; main.RegisterEvent(Main.GetActorName(akSpeaker) + " changed the sex scene to " + tags + " instead: " + newScene, "info_sexscene")
-      Return
-    EndIf
+  if useOstim
+    minai_SexOstim.StartSexOrSwitchToGroup(actors, akSpeaker, tags)
   else
-    ; Sexlab
-    Actor[] actorsInScene
-    int i = 0
-    int numMales = 0
-    int numFemales = 0
-    int threadID = slf.FindPlayerController()
-    sslThreadController Controller
-    Controller = slf.ThreadSlots.GetController(threadID)
-    if (!Controller)
-      isNewScene = true
-      actorsInScene = actors
-    else
-      actorsInScene = Controller.Positions
-    EndIf
-    while i < actorsInScene.Length
-      if actorsInScene[i] == akSpeaker && !isNewScene
-        bSpeakerInScene = True
-      EndIf
-      if actorsInScene[i] == PlayerRef
-        bPlayerInScene = True
-      EndIf
-      if slf.GetGender(actorsInScene[i]) == 0
-        numMales += 1
-      Else
-        NumFemales += 1
-      EndIf
-      if !CanAnimate(actorsInScene[i])
-        bCanAnimate = False
-      EndIf
-      i += 1
-    EndWhile
-
-    if lastTag != "" && lastTag == tags && bSpeakerInScene
-      Main.Warn("Aborting StartSexOrSwitchTo: Tag '" + tags + "' is the same as previous tag '" + lastTag +"'")
-      return
-    EndIf
+    bool skip = false
+    if(tags == lastTag)
```

Comment:
The logic regresses here - This doesn't handle new actors joining the same tag.

Reply:
restored logic in minai_SexSexlab.psc